valet/valet_suite_test.go

```diff
@@ -37,252 +38,242 @@ import (
-			foundDirs = append(foundDirs, p)
+		for path := range paths {
+			foundDirs = append(foundDirs, path)
 		}
-
-		select {
-		case err := <-errs:
+		for err := range errs {
 			Expect(err).NotTo(HaveOccurred())
-		default:
 		}
 
 		cancel()
 	})
 
-	Context("Using a prune function", func() {
+	When("using a prune function", func() {
 		It("should find paths, except those pruned", func() {
-			Expect(len(foundDirs)).To(Equal(len(expectedPaths)))
-
-			for i, ep := range expectedPaths {
-				a, err := filepath.Abs(ep)
-				Expect(err).NotTo(HaveOccurred())
-
-				fp, _ := valet.NewFilePath(a)
-
-				Expect(foundDirs[i].Location).To(Equal(fp.Location))
-				Expect(foundDirs[i].Info).ToNot(BeNil())
+			expectedPaths := []string{
+				".",
+				"1",
+				"testdir",
 			}
+
+			Expect(foundDirs).To(WithTransform(pathTransform,
+				ConsistOf(expectedPaths)))
 		})
 	})
 })
 
 var _ = Describe("Find files at intervals", func() {
 	var (
+		foundFiles    []valet.FilePath
+		pathTransform localPathTransform
+
+		dataDir = "testdata/valet"
+
 		expectedPaths = []string{
-			"testdata/valet/1/reads/fast5/reads1.fast5",
-			"testdata/valet/1/reads/fast5/reads1.fast5.md5",
-			"testdata/valet/1/reads/fast5/reads2.fast5",
-			"testdata/valet/1/reads/fast5/reads3.fast5",
-			"testdata/valet/1/reads/fastq/reads1.fastq",
-			"testdata/valet/1/reads/fastq/reads1.fastq.md5",
-			"testdata/valet/1/reads/fastq/reads2.fastq",
-			"testdata/valet/1/reads/fastq/reads2.fastq.gz",
-			"testdata/valet/1/reads/fastq/reads3.fastq",
+			"1/reads/fast5/reads1.fast5",
+			"1/reads/fast5/reads1.fast5.md5",
+			"1/reads/fast5/reads2.fast5",
+			"1/reads/fast5/reads3.fast5",
+			"1/reads/fastq/reads1.fastq",
+			"1/reads/fastq/reads1.fastq.md5",
+			"1/reads/fastq/reads2.fastq",
+			"1/reads/fastq/reads2.fastq.gz",
+			"1/reads/fastq/reads3.fastq",
 		}
-
-		foundFiles = map[string]bool{}
 	)
 
 	BeforeEach(func() {
+		absDataDir, err := filepath.Abs(dataDir)
+		Expect(err).NotTo(HaveOccurred())
+		pathTransform = makeLocalPathTransform(absDataDir)
+
 		cancelCtx, cancel := context.WithCancel(context.Background())
 		interval := 500 * time.Millisecond
 
-		paths, errs := valet.FindFilesInterval(cancelCtx, "testdata/valet",
+		paths, errs := valet.FindFilesInterval(cancelCtx, dataDir,
 			valet.IsRegular, valet.IsFalse, interval)
 
 		// Find files or timeout and cancel
-		done := make(chan bool, 2)
-
-		go func() {
-			defer cancel()
+		found := make(map[string]valet.FilePath) // FilePaths are not comparable
 
-			timer := time.NewTimer(5 * interval)
-			<-timer.C
-			done <- true // Timeout
-		}()
+		var wg sync.WaitGroup
+		wg.Add(1)
 
 		go func() {
+			defer wg.Done()
 			defer cancel()
 
-			foundFiles = make(map[string]bool) // FilePaths are not comparable
-			for p := range paths {
-				foundFiles[p.Location] = true
-				if len(foundFiles) >= len(expectedPaths) {
-					done <- true // Find files
+			for {
+				select {
+				case <-time.After(5 * interval):
```

Comment:
https://play.golang.org/p/vvDqImOu94F
You have to declare the After before the for loop, or it will be reset on each loop and and only trigger 5 * interval after the last path came from the paths channel.